scripts/Interface/reduction_gui/widgets/reflectometer/launch_peak_back_selection_1d.py

```diff
@@ -512,67 +410,161 @@ def disconnect(self):
         axes2 = self.fig.add_subplot(212)
 
         x = self.x2
         y = self.y2
 
         if bLinear:
-            _line2, = axes2.plot(x,y,color='black')
+            _line2, = axes2.plot(x, y, color='black')
         else:
-            _line2, = axes2.semilogy(x,y,color='black')
+            _line2, = axes2.semilogy(x, y, color='black')
 
-        ymax=max(y)
-        if bLinear:
-            ymin=0
-        else:
-            ymin = 1
+        ymax = max(y)
 
-        #low resolution range
+        # low resolution range
         peakx1 = self._lowresFromValue
         self.lowresFrom.setText(str(peakx1))
         peakx2 = self._lowresToValue
         self.lowresTo.setText(str(peakx2))
         if bLinear:
-            rectpeakx1 = axes2.plot([peakx1,peakx1],[ymin,ymax], '--', color='green')
-            rectpeakx2 = axes2.plot([peakx2,peakx2],[ymin,ymax], '--', color='green')
+            rectpeakx1 = axes2.plot([peakx1, peakx1], [ymin, ymax], '--', color='green')
+            rectpeakx2 = axes2.plot([peakx2, peakx2], [ymin, ymax], '--', color='green')
         else:
-            rectpeakx1 = axes2.semilogy([peakx1,peakx1],[ymin,ymax], '--', color='green')
-            rectpeakx2 = axes2.semilogy([peakx2,peakx2],[ymin,ymax], '--', color='green')
+            rectpeakx1 = axes2.semilogy([peakx1, peakx1], [ymin, ymax], '--', color='green')
+            rectpeakx2 = axes2.semilogy([peakx2, peakx2], [ymin, ymax], '--', color='green')
 
-#        self.drs2 = []
         for rect in rectpeakx1:
-            dr = DraggableRectangle(rect, 'lowres_from', self, parent)
+            dr = _DraggableRectangle(rect, 'lowres_from', self, parent)
             dr.connect()
             self.drs.append(dr)
 
         for rect in rectpeakx2:
-            dr = DraggableRectangle(rect, 'lowres_to', self, parent)
+            dr = _DraggableRectangle(rect, 'lowres_to', self, parent)
             dr.connect()
             self.drs.append(dr)
 
         self.canvas.draw()
 
-        #set the x-axes range visible
+        # set the x-axes range visible
         axes.set_xlim(self.peak_back_xlim)
         axes2.set_xlim(self.low_res_xlim)
+
+
+class _DraggableRectangle:
+    def __init__(self, rect, type, super, parent):
+        self.rect = rect
+        self.xpress = None
+        self.type = type  # peak_from, peak_to, back_from, back_to...
+        self.super = super
+        self.parent = parent
+
+    def connect(self):
+        'connect to all the events we need'
+        self.cidpress = self.rect.figure.canvas.mpl_connect('button_press_event', self.on_press)
+        self.cidrelease = self.rect.figure.canvas.mpl_connect('button_release_event', self.on_release)
+        self.cidmotion = self.rect.figure.canvas.mpl_connect('motion_notify_event', self.on_motion)
+
+    def on_press(self, event):
+        'on button press we will see if the mouse is over us and store some data'
+        if event.inaxes != self.rect.axes:
+            return
```

Comment:
flake8 should complain about the 2 blank lines in this function

Reply:
I only see 1 blank line?

Reply:
yes, you are right. sorry about that.